tests/unit/CoreQuoteTest.php

```diff
@@ -0,0 +1,245 @@
+			<!-- wp:quote {"className":"custom-quote-class"} -->
+			<blockquote class="wp-block-quote custom-quote-class"><p>This is a sample quote block content.</p><cite>Author Name</cite></blockquote>
+			<!-- /wp:quote -->
+		';
+
+		// Update the post content with the block content.
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$query     = $this->query();
+		$variables = [
+			'id' => $this->post_id,
+		];
+
+		$actual = graphql( compact( 'query', 'variables' ) );
+
+		$this->assertArrayNotHasKey( 'errors', $actual, 'There should not be any errors' );
+		$this->assertArrayHasKey( 'data', $actual, 'The data key should be present' );
+		$this->assertArrayHasKey( 'post', $actual['data'], 'The post key should be present' );
+		$this->assertEquals( $this->post_id, $actual['data']['post']['databaseId'], 'The post ID should match' );
+		$this->assertEquals( 1, count( $actual['data']['post']['editorBlocks'] ), 'There should be only one block' );
+
+		$block = $actual['data']['post']['editorBlocks'][0];
+		$this->assertNotEmpty( $block['apiVersion'], 'The apiVersion should be present' );
+		$this->assertEquals( 'text', $block['blockEditorCategoryName'], 'The blockEditorCategoryName should be text' );
+		$this->assertNotEmpty( $block['clientId'], 'The clientId should be present' );
+		$this->assertEquals( $block['cssClassNames'][0], 'custom-quote-class' );
+		$this->assertEmpty( $block['innerBlocks'], 'There should be no inner blocks' );
+		$this->assertEquals( 'core/quote', $block['name'], 'The block name should be core/quote' );
+		$this->assertEmpty( $block['parentClientId'], 'There should be no parentClientId' );
+		$this->assertNotEmpty( $block['renderedHtml'], 'The renderedHtml should be present' );
+
+		// Verify the attributes.
+		$this->assertEquals(
+			[
+				'anchor'          => null,
+				'backgroundColor' => null,
+				'citation'        => 'Author Name',
+				'className'       => 'custom-quote-class',
+				'cssClassName'    => 'wp-block-quote custom-quote-class',
+				'fontFamily'      => null,
+				'fontSize'        => null,
+				'gradient'        => null,
+				'lock'            => null,
+				'style'           => null,
+				'textColor'       => null,
+				'value'           => '<p>This is a sample quote block content.</p>',
+			],
+			$block['attributes'],
+		);
+	}
+
+	/**
+	 * Test case for retrieving core quote block untested attributes.
+	 *
+	 * The following aspects are tested:
+	 * - The absence of errors in the GraphQL response.
+	 * - Presence of the 'data' and 'post' keys in the response.
+	 * - Matching post ID.
+	 * - Correct block name ('core/quote').
+	 * - Correct retrieval of the block's attributes, especially:
+	 *   - 'anchor'
+	 *   - 'backgroundColor'
+	 *   - 'fontFamily'
+	 *   - 'fontSize'
+	 *   - 'gradient'
+	 *   - 'lock'
+	 *   - 'style'
+	 *   - 'textColor'
+	 *
+	 * @return void
+	 */
+	public function test_retrieve_core_quote_attributes() {
```

Comment:
If there's so many attributes you cant give this a self-describing name, then use `test_retrieve_core_quote_additional_attributes()`

Reply:
Fixed ✅